source/MaterialXCore/Document.cpp

```diff
@@ -20,7 +20,9 @@ NodeDefPtr getShaderNodeDef(ElementPtr shaderRef)
     if (shaderRef->hasAttribute(NodeDef::NODE_DEF_ATTRIBUTE))
     {
         string nodeDefString = shaderRef->getAttribute(NodeDef::NODE_DEF_ATTRIBUTE);
-        return shaderRef->resolveRootNameReference<NodeDef>(nodeDefString);
+        ConstElementPtr root = shaderRef->getRoot();
```

Comment:
Since we're now writing out this code explicitly, I would recommend calling `DocumentPtr doc = shaderRef->getDocument()` here, which would allow us to call `doc->getNodeDef()` in the two calls below.

Reply:
Unfortunately, `Document::getNodeDef()` is not equivalent to resolve logic as it does not try and use the qualified name. I don't think I want to change that method and am hesitant to add a new method just for this small change.